### DISCUSSION TITLE
Show branch name in workflow run history

### DISCUSSION THREAD

**Discussion by AlexandruAIlea** [community/community#44468] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello,

I've been searching the answer on the Github documentation but had no success.

I'm working on a project which defines 2 actions.
Both of them are using workflow_dispatch so that we can start the workflow manually from a branch.

When looking at the workflow history, one of the workflows shows the branch name for each run and the other one is not.
Just like in this documentation link from Github, screenshot from step 3 shows the branch name and the one from step 4 doesn't show it:

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/viewing-workflow-run-history

How to configure the workflow to show the branch for each run?

Thanks,
Alex

**Comment by jge162** [comment]
In order to configure a workflow to show the branch name for each run, you will need to make sure that the branch name is being passed as an input to the workflow when it is manually triggered using the workflow_dispatch event. You can do this by adding the branch parameter to the inputs section of your workflow file and then passing the appropriate branch name when manually triggering the workflow. For example, in your workflow file you would add branch: ${{ inputs.branch }} and when manually triggering the workflow, you would pass the branch input with a value of the branch you want to run the workflow on.

**Reply by AlexandruAIlea** [reply]
@jge162 Thanks for helping me on this.

I have changed the workflow like you suggested, but it still didn't add the branch in the history. Also if I add the branch in the parameter I will have the branch input twice: the dropdown that is by default added by Github and the one I added in the workflow file. I would prefer if possible to use the default dropdown only.

Also just figured out that on the workflow where the branch name is shown in the list, it is actually shown for the workflows that were run on events (e.g. push), not on manual runs. My original question is whether it's possible to see the branch for manual runs since this workflow that I want to make it work is currently designed to be run only manually.

Once again, I appreciate your help @jge162 !

**Reply by AlexandruAIlea** [reply]
actually I just found this open thread now after figuring out this only happens on manual run:

https://github.com/community/community/discussions/15690

**Reply by jge162** [reply]
Hm. How about this? For manual runs, it's worth noting that the branch name is typically only available for events that are associated with a specific branch, such as a push event. For manual runs, the branch name may not be available by default. However, you could try passing the branch name as an input to the workflow when it is run manually. You can use the with keyword to pass inputs to a workflow, for example:


workflow_dispatch = github.workflows.create_dispatch("my_workflow", ref="my-branch", inputs={...})

**Reply by AlexandruAIlea** [reply]
@jge162 I have added the with keyword like this:

      - name: Check out code
        uses: actions/checkout@v3
        with:
          ref: ${{ github.event.inputs.branch }}"

The keyword is working such that it is checking out the code from the branch I'm choosing. 
However it still won't show the branch in the workflow runs list.
Note that I didn't add the branch name as an input to avoid having this parameter twice.

**Reply by jge162** [reply]
see below. Typo at the end of branch? On text above?

- name: Check out code
  uses: actions/checkout@v3
  with:
    ref: ${{ github.event.inputs.branch }}
  id: checkout

- name: Set branch output
  run: echo ::set-output name=branch::${{ steps.checkout.outputs.ref }}

**Reply by AlexandruAIlea** [reply]
Thanks for noticing the typo... That's fixed now.

I have added the step to set the output from the branch, but it's still not showing the branch in the workflow history.

**Reply by jge162** [reply]
If the "ref" output is not being set correctly, you may need to check the configuration of the "actions/checkout@v3" action, and make sure that the "ref" parameter is set to the correct value.

Another possibility is that the "branch" output is not being referenced correctly in the next steps. You should make sure that the reference to the output is correct,

 it should be 
${{ steps.set_branch.outputs.branch }} 

instead of 

${{ steps.checkout.outputs.ref }}.

It's also important to check if the syntax of the "echo" command is correct and the string ::set-output name=branch:: is being used correctly.

**Reply by jge162** [reply]
You can try to check if the "ref" output is set correctly by adding a step to print its value, like this:

- name: Print ref output
  run: echo ${{ steps.checkout.outputs.ref }}

**Reply by jge162** [reply]
@AlexandruAIlea 

it should be
${{ steps.set_branch.outputs.branch }}

instead of

${{ steps.checkout.outputs.ref }}.

**Comment by jsoref** [comment, marked as answer]
This should just be closed as a duplicate of #15690

**Reply by AlexandruAIlea** [reply]
@jsoref thanks for all your help provided for this discussion! I will close it as according to the discussion you referred it is a known issue.
